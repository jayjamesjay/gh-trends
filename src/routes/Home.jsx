import React, { useCallback, useEffect, useMemo } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import getAndSave, { Url, defApi, perPage, addLang, load } from '../components/Fetch';
import { ViewSingle } from '../components/View';
import { queries, languages } from '../components/Data';
import RepoInfoList from '../components/RepoInfoList';
import RepoInfo from '../components/RepoInfo';
import { H1 } from '../styles/Headers';
import { FormAlt } from '../styles/Form';
import Select from '../components/Select';
import { save } from '../actions';

const langs = ['All Languages', ...Object.keys(languages)];
const timeList = Object.keys(queries);

const mapStateToProps = state => {
  return {
    saved: state.saved
  };
};

const mapDispatchToProps = { save };

// eslint-disable-next-line no-shadow
export function Home({ saved, save }) {
  const [lang, setLang] = React.useState('All');
  const [time, setTime] = React.useState('This Week');
  const [repoInfo, setRepoInfo] = React.useState(new RepoInfoList([], 1));
  const [loading, setLoading] = React.useState(load.INPROGRESS);
  const abortController = new AbortController();
  const { signal } = abortController;
  const query = useMemo(() => addLang(queries[time], lang), [lang, time]);

  const onSelect = useCallback((event, callback) => {
    const val = event.target.value;
    callback(val);
  }, []);

  const selectLang = useCallback(
    event => {
      onSelect(event, setLang);
    },
    [onSelect, setLang]
  );

  const selectTime = useCallback(
    event => {
      onSelect(event, setTime);
    },
    [onSelect, setTime]
  );

  const loadData = useCallback(
    (currData = [], currPage = 1) => {
      setLoading(load.INPROGRESS);
      const preUrl = new Url(defApi).query(query).parts(perPage);
      const infoList = new RepoInfoList(currData, currPage);
      return getAndSave(infoList, preUrl, signal, setRepoInfo)
        .then(() => {
          setLoading(load.LOADED);
        })
        .catch(() => {
          setLoading(load.ERORR);
        });
    },
    [query, signal, setRepoInfo]
  );

  useEffect(() => {
    loadData();

    return () => {
      abortController.abort();
    };
  }, [lang, time]); // eslint-disable-line

  return (
    <>
      <header>
        <H1>Trending repositories</H1>
      </header>
      <FormAlt>
        <Select curr={lang} onSelect={selectLang} options={langs} label="Languages" />
        <Select curr={time} onSelect={selectTime} options={timeList} label="Periods of time" />
      </FormAlt>
      <ViewSingle
        data={repoInfo.data}
        loadData={() => loadData(repoInfo.data, repoInfo.page)}
        save={save}
        saved={saved}
        loading={loading}
      />
    </>
  );
}

Home.propTypes = {
  saved: PropTypes.arrayOf(PropTypes.instanceOf(RepoInfo)).isRequired,
  save: PropTypes.func.isRequired
};

const HomeContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(Home);

export default HomeContainer;
