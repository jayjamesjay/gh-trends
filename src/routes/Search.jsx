import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ViewSingle } from '../components/View';
import getJSON, { requestUrl, api, perPage, query } from '../components/Fetch';
import RepoInfoList, { RepoInfo, languages, addLang } from '../components/Data';
import { ButtonIcon } from '../styles/Button';
import { TextInput, Form, FormGroup } from '../styles/Form';
import { Img } from '../styles/Img';
import SelectLang from '../components/SelectLang';

export default class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      search: '',
      lang: 'all',
      data: [],
      page: 1
    };
  }

  loadData = () => {
    const { data, page } = this.state;
    this.makeRequest(data, page);
  };

  reloadData = () => {
    this.makeRequest([], 1);
  };

  makeRequest = (data, page) => {
    const { search } = this.state;
    let currData = data;
    const argsList = [api, query(search), perPage];
    if (page > 1) {
      argsList.push(`page=${page + 1}`);
    }

    const url = requestUrl(...argsList);

    getJSON(url)
      .then(result => {
        currData = currData.concat(RepoInfoList.fromGithubRes(result.items));
        const currPage = page + 1;

        this.setState({
          data: currData,
          page: currPage
        });
      })
      .catch(_ => {
        this.asyncRequest = null;
      });
  };

  onInput = event => {
    const search = event.target.value;
    this.setState({ search });
  };

  onSubmit = event => {
    event.preventDefault();
  };

  onKeyPress = event => {
    if (event.key === 'Enter') {
      this.reloadData();
    }
  };

  onSelect = event => {
    const lang = event.target.value;
    const { search } = this.state;
    const newSearch = addLang(search.slice(), lang);

    this.setState({ lang, search: newSearch });
  };

  render() {
    const {
      onInput,
      onSubmit,
      onKeyPress,
      onSelect,
      loadData,
      reloadData,
      state: { search, data, lang },
      props: { save, saved }
    } = this;

    return (
      <>
        <Form onSubmit={onSubmit} noValidate>
          <SelectLang
            curr={lang}
            onSelect={onSelect}
            languages={Object.keys(languages)}
            label="Add language"
          />
          <FormGroup>
            <TextInput
              aria-label="Search for repositories"
              required
              type="text"
              onChange={onInput}
              value={search}
              onKeyPress={onKeyPress}
            />
            <ButtonIcon onClick={reloadData}>
              <Img src="./assets/img/search.svg" alt="Search" />
            </ButtonIcon>
          </FormGroup>
        </Form>
        <ViewSingle data={data} loadData={loadData} save={save} saved={saved} />
      </>
    );
  }
}

Search.propTypes = {
  save: PropTypes.func.isRequired,
  saved: PropTypes.arrayOf(PropTypes.instanceOf(RepoInfo)).isRequired
};
