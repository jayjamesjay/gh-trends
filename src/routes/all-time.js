import React, { Component } from "react";
import { DataSquare } from "../components/main";

export default class AllTime extends Component {
  render() {
    return (
      <DataSquare
        title="All Time"
        content="Odio ut sem nulla pharetra diam sit. Ipsum dolor sit amet 
          consectetur adipiscing elit pellentesque habitant. Mattis aliquam 
          faucibus purus in massa tempor nec feugiat nisl. Hac habitasse platea 
          dictumst vestibulum rhoncus est. Adipiscing tristique risus nec feugiat. 
          Sed blandit libero volutpat sed cras. Interdum consectetur libero id 
          faucibus nisl tincidunt. Tristique senectus et netus et malesuada. 
          Vitae tempus quam pellentesque nec nam aliquam sem et. Semper feugiat 
          nibh sed pulvinar proin gravida hendrerit lectus a. Fusce ut placerat 
          orci nulla pellentesque. Eget magna fermentum iaculis eu non. Tellus 
          integer feugiat scelerisque varius morbi. Duis ut diam quam nulla. 
          In iaculis nunc sed augue."
      />
    );
  }
}
