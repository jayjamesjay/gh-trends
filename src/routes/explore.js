import React, { Component } from "react";
import { InfoBlock } from "../components/main";

//Temporarily added testing data
const dataExplore = {
  data: {
    search: {
      nodes: [
        {
          nameWithOwner: "imfunniee/gitfolio",
          url: "https://github.com/imfunniee/gitfolio",
          description:
            ":octocat: personal website + blog  for every github user",
          stargazers: {
            totalCount: 3124
          },
          repositoryTopics: {
            nodes: [
              {
                topic: {
                  name: "personal-website"
                }
              },
              {
                topic: {
                  name: "awesome"
                }
              },
              {
                topic: {
                  name: "node"
                }
              }
            ]
          },
          languages: {
            nodes: [
              {
                name: "HTML",
                color: "#e34c26"
              }
            ]
          }
        },
        {
          nameWithOwner: "nokia/memory-profiler",
          url: "https://github.com/nokia/memory-profiler",
          description: null,
          stargazers: {
            totalCount: 1624
          },
          repositoryTopics: {
            nodes: []
          },
          languages: {
            nodes: [
              {
                name: "Shell",
                color: "#89e051"
              }
            ]
          }
        }
      ]
    }
  }
};

const data = dataExplore.data.search.nodes;

export default class Explore extends Component {
  render() {
    return (
      <>
        {data.map((node, i) => (
          <InfoBlock key={node.nameWithOwner} info={node} />
        ))}
      </>
    );
  }
}
