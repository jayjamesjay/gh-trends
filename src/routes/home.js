import React, { Component } from 'react';
import getJSON, {
  requestUrl,
  api,
  order,
  sort,
  perPage,
  query,
} from '../components/fetch';
import { InfoBlock, Tabs } from '../components/main';
import { Content, ButtonMain } from '../components/styles';

import Data, { queryList, initData } from '../components/data';

export default class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: [
        new Data('Week', initData, 1),
        new Data('Month', initData, 1),
        new Data('All Time', initData, 1),
      ],
    };
  }

  componentDidMount() {
    const newData = this.state.data.slice();
    const idList = ['Week', 'Month', 'All Time'];

    for (let i = 0; i < newData.length; i++) {
      const url = requestUrl(api, query(queryList[i]), sort, order, perPage);

      this._asyncRequest = getJSON(url).then(result => {
        this._asyncRequest = null;

        newData[i] = new Data(idList[i], Data.fromGithubRes(result.items), 1);

        this.setState({
          data: newData,
        });
      });
    }
  }

  componentWillUnmount() {
    if (this._asyncRequest) {
      this._asyncRequest.cancel();
    }
  }

  loadData = id => {
    const currData = this.state.data;
    const idx = currData.findIndex(elem => elem.id === id);
    const url = requestUrl(
      api,
      query(queryList[idx]),
      sort,
      order,
      perPage,
      'page=' + (currData[idx].page + 1)
    );

    getJSON(url).then(result => {
      currData[idx].data = currData[idx].data.concat(
        Data.fromGithubRes(result.items)
      );
      currData[idx].page += 1;

      this.setState({
        data: currData,
      });
    });
  };

  render() {
    return (
      <Tabs>
        {this.state.data.map(elem => {
          return (
            <div key={elem.id} label={elem.id}>
              <View id={elem.id} data={elem.data} loadData={this.loadData} />
            </div>
          );
        })}
      </Tabs>
    );
  }
}

export class View extends Component {
  onClick = () => {
    this.props.loadData(this.props.id);
  };

  render() {
    return (
      <>
        <Content>
          {this.props.data.map((node, id) => (
            <InfoBlock key={id} info={node} />
          ))}
        </Content>
        <footer>
          <ButtonMain onClick={this.onClick}>Show more</ButtonMain>
        </footer>
      </>
    );
  }
}
