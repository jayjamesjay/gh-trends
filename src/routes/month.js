import React, { Component } from "react";
import StandarView from "../components/main";
import formatDate from "../components/date";

const monthAgo = new Date();
monthAgo.setMonth(monthAgo.getMonth() - 1);
const monthAgoStr = formatDate(monthAgo);

const query = `sort:stars-desc created:>2019-05-02`;
const title = "Trending repos this month";
const description = "Repositories created this month with biggest amount of stars."

export default class Month extends Component {
  render() {
    return <StandarView query={query} title={title} description={description}/>;
  }
}
