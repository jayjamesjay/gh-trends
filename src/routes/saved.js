import React, { Component } from "react";

export default class Month extends Component {
  render() {
    return <h1>Work in Progress...</h1>;
  }
}
