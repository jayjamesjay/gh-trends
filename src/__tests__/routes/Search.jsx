import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import { Search } from '../../routes/Search';

describe('<Search />', () => {
  const func = () => {};
  const query = 'react';
  const lang = 'JavaScript';

  const wrapper = () => render(<Search save={func} saved={[]} />);
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation((init) => [init, setState]);

  afterEach(() => {
    jest.clearAllMocks();
  });

  it(`renders with default style`, () => {
    const component = renderer.create(<Search save={func} saved={[]} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('onInput', () => {
    wrapper();
    fireEvent.input(screen.getByLabelText('Search for repositories'), { target: { value: query } });

    expect(setState).toHaveBeenCalledWith(query);
  });

  it('onSelect', () => {
    wrapper();
    fireEvent.change(screen.getByLabelText('Languages'), { target: { value: lang } });

    expect(setState).toHaveBeenCalledWith(lang);
  });
});
