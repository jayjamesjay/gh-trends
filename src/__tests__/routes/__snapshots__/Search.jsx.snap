// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders with default style 1`] = `
[
  .c0 {
  margin: 3rem auto 1rem auto;
  font-size: 2rem;
  text-transform: uppercase;
}

@media screen and (min-width:40rem) {

}

<header>
    <h1
      className="c0"
    >
      Search for repositories
    </h1>
  </header>,
  .c7 {
  cursor: pointer;
  width: 2rem;
  height: 2rem;
  background: transparent;
  border: 0;
}

.c2 {
  width: 100%;
  height: auto;
  filter: invert(0%);
}

.c3 {
  width: 3rem;
}

.c0 {
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  max-width: 90%;
}

.c1 {
  padding: 1rem;
}

.c6 {
  min-width: 50%;
  padding: 0.5rem;
  background: #fafafa;
  border: solid #767676;
  border-width: 0 0 0.15rem 0;
  font-size: 1.2rem;
  color: #000;
}

.c6:focus {
  border: solid #000;
  border-width: 0 0 0.15rem 0;
}

.c4 {
  max-width: 80%;
  margin: 0.5rem;
  padding: 0.5rem;
  background: #fff;
  border: none;
  border-radius: 0.5rem;
  box-shadow: 0px 3px 5px 1px rgba(217,217,217,0.82);
  font-size: 1.3rem;
  color: #000;
  text-align: center;
  text-transform: uppercase;
  appearance: none;
}

.c5 {
  font-size: 1.2rem;
  line-break: loose;
  text-align: center;
  text-transform: uppercase;
}

@media screen and (min-width:40rem) {

}

<form
    className="c0 c1"
    noValidate={true}
    onSubmit={[Function]}
  >
    <img
      alt="Add language"
      className="c2 c3"
      src="test-file-stub"
      title="Add language"
    />
    <select
      aria-label="Languages"
      className="c4"
      onChange={[Function]}
      value="all"
    >
      <option
        className="c5"
        value="All Languages"
      >
        All Languages
      </option>
      <option
        className="c5"
        value="1C Enterprise"
      >
        1C Enterprise
      </option>
      <option
        className="c5"
        value="ABAP"
      >
        ABAP
      </option>
      <option
        className="c5"
        value="AGS Script"
      >
        AGS Script
      </option>
      <option
        className="c5"
        value="AMPL"
      >
        AMPL
      </option>
      <option
        className="c5"
        value="ANTLR"
      >
        ANTLR
      </option>
      <option
        className="c5"
        value="API Blueprint"
      >
        API Blueprint
      </option>
      <option
        className="c5"
        value="APL"
      >
        APL
      </option>
      <option
        className="c5"
        value="ASP"
      >
        ASP
      </option>
      <option
        className="c5"
        value="ATS"
      >
        ATS
      </option>
      <option
        className="c5"
        value="ActionScript"
      >
        ActionScript
      </option>
      <option
        className="c5"
        value="Ada"
      >
        Ada
      </option>
      <option
        className="c5"
        value="Agda"
      >
        Agda
      </option>
      <option
        className="c5"
        value="Alloy"
      >
        Alloy
      </option>
      <option
        className="c5"
        value="AngelScript"
      >
        AngelScript
      </option>
      <option
        className="c5"
        value="AppleScript"
      >
        AppleScript
      </option>
      <option
        className="c5"
        value="Arc"
      >
        Arc
      </option>
      <option
        className="c5"
        value="AspectJ"
      >
        AspectJ
      </option>
      <option
        className="c5"
        value="Assembly"
      >
        Assembly
      </option>
      <option
        className="c5"
        value="AutoHotkey"
      >
        AutoHotkey
      </option>
      <option
        className="c5"
        value="AutoIt"
      >
        AutoIt
      </option>
      <option
        className="c5"
        value="Ballerina"
      >
        Ballerina
      </option>
      <option
        className="c5"
        value="Batchfile"
      >
        Batchfile
      </option>
      <option
        className="c5"
        value="BlitzMax"
      >
        BlitzMax
      </option>
      <option
        className="c5"
        value="Boo"
      >
        Boo
      </option>
      <option
        className="c5"
        value="Brainfuck"
      >
        Brainfuck
      </option>
      <option
        className="c5"
        value="C"
      >
        C
      </option>
      <option
        className="c5"
        value="C#"
      >
        C#
      </option>
      <option
        className="c5"
        value="CSS"
      >
        CSS
      </option>
      <option
        className="c5"
        value="Ceylon"
      >
        Ceylon
      </option>
      <option
        className="c5"
        value="Chapel"
      >
        Chapel
      </option>
      <option
        className="c5"
        value="Cirru"
      >
        Cirru
      </option>
      <option
        className="c5"
        value="Clarion"
      >
        Clarion
      </option>
      <option
        className="c5"
        value="Clean"
      >
        Clean
      </option>
      <option
        className="c5"
        value="Click"
      >
        Click
      </option>
      <option
        className="c5"
        value="Clojure"
      >
        Clojure
      </option>
      <option
        className="c5"
        value="CoffeeScript"
      >
        CoffeeScript
      </option>
      <option
        className="c5"
        value="ColdFusion"
      >
        ColdFusion
      </option>
      <option
        className="c5"
        value="Common Lisp"
      >
        Common Lisp
      </option>
      <option
        className="c5"
        value="Common Workflow Language"
      >
        Common Workflow Language
      </option>
      <option
        className="c5"
        value="Component Pascal"
      >
        Component Pascal
      </option>
      <option
        className="c5"
        value="Crystal"
      >
        Crystal
      </option>
      <option
        className="c5"
        value="C++"
      >
        C++
      </option>
      <option
        className="c5"
        value="Cuda"
      >
        Cuda
      </option>
      <option
        className="c5"
        value="D"
      >
        D
      </option>
      <option
        className="c5"
        value="DM"
      >
        DM
      </option>
      <option
        className="c5"
        value="Dart"
      >
        Dart
      </option>
      <option
        className="c5"
        value="DataWeave"
      >
        DataWeave
      </option>
      <option
        className="c5"
        value="Dogescript"
      >
        Dogescript
      </option>
      <option
        className="c5"
        value="Dylan"
      >
        Dylan
      </option>
      <option
        className="c5"
        value="E"
      >
        E
      </option>
      <option
        className="c5"
        value="ECL"
      >
        ECL
      </option>
      <option
        className="c5"
        value="EQ"
      >
        EQ
      </option>
      <option
        className="c5"
        value="Eiffel"
      >
        Eiffel
      </option>
      <option
        className="c5"
        value="Elixir"
      >
        Elixir
      </option>
      <option
        className="c5"
        value="Elm"
      >
        Elm
      </option>
      <option
        className="c5"
        value="Emacs Lisp"
      >
        Emacs Lisp
      </option>
      <option
        className="c5"
        value="EmberScript"
      >
        EmberScript
      </option>
      <option
        className="c5"
        value="Erlang"
      >
        Erlang
      </option>
      <option
        className="c5"
        value="F#"
      >
        F#
      </option>
      <option
        className="c5"
        value="FLUX"
      >
        FLUX
      </option>
      <option
        className="c5"
        value="Factor"
      >
        Factor
      </option>
      <option
        className="c5"
        value="Fancy"
      >
        Fancy
      </option>
      <option
        className="c5"
        value="Fantom"
      >
        Fantom
      </option>
      <option
        className="c5"
        value="Forth"
      >
        Forth
      </option>
      <option
        className="c5"
        value="Fortran"
      >
        Fortran
      </option>
      <option
        className="c5"
        value="FreeMarker"
      >
        FreeMarker
      </option>
      <option
        className="c5"
        value="Frege"
      >
        Frege
      </option>
      <option
        className="c5"
        value="Game Maker Language"
      >
        Game Maker Language
      </option>
      <option
        className="c5"
        value="Genie"
      >
        Genie
      </option>
      <option
        className="c5"
        value="Gherkin"
      >
        Gherkin
      </option>
      <option
        className="c5"
        value="Glyph"
      >
        Glyph
      </option>
      <option
        className="c5"
        value="Gnuplot"
      >
        Gnuplot
      </option>
      <option
        className="c5"
        value="Go"
      >
        Go
      </option>
      <option
        className="c5"
        value="Golo"
      >
        Golo
      </option>
      <option
        className="c5"
        value="Gosu"
      >
        Gosu
      </option>
      <option
        className="c5"
        value="Grammatical Framework"
      >
        Grammatical Framework
      </option>
      <option
        className="c5"
        value="Groovy"
      >
        Groovy
      </option>
      <option
        className="c5"
        value="HTML"
      >
        HTML
      </option>
      <option
        className="c5"
        value="Hack"
      >
        Hack
      </option>
      <option
        className="c5"
        value="Harbour"
      >
        Harbour
      </option>
      <option
        className="c5"
        value="Haskell"
      >
        Haskell
      </option>
      <option
        className="c5"
        value="Haxe"
      >
        Haxe
      </option>
      <option
        className="c5"
        value="Hy"
      >
        Hy
      </option>
      <option
        className="c5"
        value="IDL"
      >
        IDL
      </option>
      <option
        className="c5"
        value="Idris"
      >
        Idris
      </option>
      <option
        className="c5"
        value="Io"
      >
        Io
      </option>
      <option
        className="c5"
        value="Ioke"
      >
        Ioke
      </option>
      <option
        className="c5"
        value="Isabelle"
      >
        Isabelle
      </option>
      <option
        className="c5"
        value="J"
      >
        J
      </option>
      <option
        className="c5"
        value="JSONiq"
      >
        JSONiq
      </option>
      <option
        className="c5"
        value="Java"
      >
        Java
      </option>
      <option
        className="c5"
        value="JavaScript"
      >
        JavaScript
      </option>
      <option
        className="c5"
        value="Jolie"
      >
        Jolie
      </option>
      <option
        className="c5"
        value="Julia"
      >
        Julia
      </option>
      <option
        className="c5"
        value="Jupyter Notebook"
      >
        Jupyter Notebook
      </option>
      <option
        className="c5"
        value="KRL"
      >
        KRL
      </option>
      <option
        className="c5"
        value="Kotlin"
      >
        Kotlin
      </option>
      <option
        className="c5"
        value="LFE"
      >
        LFE
      </option>
      <option
        className="c5"
        value="LLVM"
      >
        LLVM
      </option>
      <option
        className="c5"
        value="LOLCODE"
      >
        LOLCODE
      </option>
      <option
        className="c5"
        value="LSL"
      >
        LSL
      </option>
      <option
        className="c5"
        value="Lasso"
      >
        Lasso
      </option>
      <option
        className="c5"
        value="Lex"
      >
        Lex
      </option>
      <option
        className="c5"
        value="LiveScript"
      >
        LiveScript
      </option>
      <option
        className="c5"
        value="LookML"
      >
        LookML
      </option>
      <option
        className="c5"
        value="Lua"
      >
        Lua
      </option>
      <option
        className="c5"
        value="MAXScript"
      >
        MAXScript
      </option>
      <option
        className="c5"
        value="MQL4"
      >
        MQL4
      </option>
      <option
        className="c5"
        value="MQL5"
      >
        MQL5
      </option>
      <option
        className="c5"
        value="MTML"
      >
        MTML
      </option>
      <option
        className="c5"
        value="Makefile"
      >
        Makefile
      </option>
      <option
        className="c5"
        value="Mask"
      >
        Mask
      </option>
      <option
        className="c5"
        value="Matlab"
      >
        Matlab
      </option>
      <option
        className="c5"
        value="Max"
      >
        Max
      </option>
      <option
        className="c5"
        value="Mercury"
      >
        Mercury
      </option>
      <option
        className="c5"
        value="Meson"
      >
        Meson
      </option>
      <option
        className="c5"
        value="Metal"
      >
        Metal
      </option>
      <option
        className="c5"
        value="Mirah"
      >
        Mirah
      </option>
      <option
        className="c5"
        value="NCL"
      >
        NCL
      </option>
      <option
        className="c5"
        value="Nearley"
      >
        Nearley
      </option>
      <option
        className="c5"
        value="Nemerle"
      >
        Nemerle
      </option>
      <option
        className="c5"
        value="NetLinx"
      >
        NetLinx
      </option>
      <option
        className="c5"
        value="NetLinx+ERB"
      >
        NetLinx+ERB
      </option>
      <option
        className="c5"
        value="NetLogo"
      >
        NetLogo
      </option>
      <option
        className="c5"
        value="NewLisp"
      >
        NewLisp
      </option>
      <option
        className="c5"
        value="Nextflow"
      >
        Nextflow
      </option>
      <option
        className="c5"
        value="Nim"
      >
        Nim
      </option>
      <option
        className="c5"
        value="Nit"
      >
        Nit
      </option>
      <option
        className="c5"
        value="Nix"
      >
        Nix
      </option>
      <option
        className="c5"
        value="Nu"
      >
        Nu
      </option>
      <option
        className="c5"
        value="OCaml"
      >
        OCaml
      </option>
      <option
        className="c5"
        value="Objective-C"
      >
        Objective-C
      </option>
      <option
        className="c5"
        value="Objective-C++"
      >
        Objective-C++
      </option>
      <option
        className="c5"
        value="Objective-J"
      >
        Objective-J
      </option>
      <option
        className="c5"
        value="Omgrofl"
      >
        Omgrofl
      </option>
      <option
        className="c5"
        value="Opal"
      >
        Opal
      </option>
      <option
        className="c5"
        value="Oxygene"
      >
        Oxygene
      </option>
      <option
        className="c5"
        value="Oz"
      >
        Oz
      </option>
      <option
        className="c5"
        value="P4"
      >
        P4
      </option>
      <option
        className="c5"
        value="PAWN"
      >
        PAWN
      </option>
      <option
        className="c5"
        value="PHP"
      >
        PHP
      </option>
      <option
        className="c5"
        value="PLSQL"
      >
        PLSQL
      </option>
      <option
        className="c5"
        value="Pan"
      >
        Pan
      </option>
      <option
        className="c5"
        value="Papyrus"
      >
        Papyrus
      </option>
      <option
        className="c5"
        value="Parrot"
      >
        Parrot
      </option>
      <option
        className="c5"
        value="Pascal"
      >
        Pascal
      </option>
      <option
        className="c5"
        value="Pep8"
      >
        Pep8
      </option>
      <option
        className="c5"
        value="Perl"
      >
        Perl
      </option>
      <option
        className="c5"
        value="Perl 6"
      >
        Perl 6
      </option>
      <option
        className="c5"
        value="PigLatin"
      >
        PigLatin
      </option>
      <option
        className="c5"
        value="Pike"
      >
        Pike
      </option>
      <option
        className="c5"
        value="PogoScript"
      >
        PogoScript
      </option>
      <option
        className="c5"
        value="PostScript"
      >
        PostScript
      </option>
      <option
        className="c5"
        value="PowerBuilder"
      >
        PowerBuilder
      </option>
      <option
        className="c5"
        value="PowerShell"
      >
        PowerShell
      </option>
      <option
        className="c5"
        value="Processing"
      >
        Processing
      </option>
      <option
        className="c5"
        value="Prolog"
      >
        Prolog
      </option>
      <option
        className="c5"
        value="Propeller Spin"
      >
        Propeller Spin
      </option>
      <option
        className="c5"
        value="Puppet"
      >
        Puppet
      </option>
      <option
        className="c5"
        value="PureBasic"
      >
        PureBasic
      </option>
      <option
        className="c5"
        value="PureScript"
      >
        PureScript
      </option>
      <option
        className="c5"
        value="Python"
      >
        Python
      </option>
      <option
        className="c5"
        value="QML"
      >
        QML
      </option>
      <option
        className="c5"
        value="R"
      >
        R
      </option>
      <option
        className="c5"
        value="RAML"
      >
        RAML
      </option>
      <option
        className="c5"
        value="RUNOFF"
      >
        RUNOFF
      </option>
      <option
        className="c5"
        value="Racket"
      >
        Racket
      </option>
      <option
        className="c5"
        value="Ragel"
      >
        Ragel
      </option>
      <option
        className="c5"
        value="Rascal"
      >
        Rascal
      </option>
      <option
        className="c5"
        value="Rebol"
      >
        Rebol
      </option>
      <option
        className="c5"
        value="Red"
      >
        Red
      </option>
      <option
        className="c5"
        value="Ren'Py"
      >
        Ren'Py
      </option>
      <option
        className="c5"
        value="Ring"
      >
        Ring
      </option>
      <option
        className="c5"
        value="Roff"
      >
        Roff
      </option>
      <option
        className="c5"
        value="Rouge"
      >
        Rouge
      </option>
      <option
        className="c5"
        value="Ruby"
      >
        Ruby
      </option>
      <option
        className="c5"
        value="Rust"
      >
        Rust
      </option>
      <option
        className="c5"
        value="SAS"
      >
        SAS
      </option>
      <option
        className="c5"
        value="SQF"
      >
        SQF
      </option>
      <option
        className="c5"
        value="SRecode Template"
      >
        SRecode Template
      </option>
      <option
        className="c5"
        value="SaltStack"
      >
        SaltStack
      </option>
      <option
        className="c5"
        value="Scala"
      >
        Scala
      </option>
      <option
        className="c5"
        value="Scheme"
      >
        Scheme
      </option>
      <option
        className="c5"
        value="Self"
      >
        Self
      </option>
      <option
        className="c5"
        value="Shell"
      >
        Shell
      </option>
      <option
        className="c5"
        value="Shen"
      >
        Shen
      </option>
      <option
        className="c5"
        value="Slash"
      >
        Slash
      </option>
      <option
        className="c5"
        value="Smalltalk"
      >
        Smalltalk
      </option>
      <option
        className="c5"
        value="Solidity"
      >
        Solidity
      </option>
      <option
        className="c5"
        value="SourcePawn"
      >
        SourcePawn
      </option>
      <option
        className="c5"
        value="Squirrel"
      >
        Squirrel
      </option>
      <option
        className="c5"
        value="Stan"
      >
        Stan
      </option>
      <option
        className="c5"
        value="Standard ML"
      >
        Standard ML
      </option>
      <option
        className="c5"
        value="SuperCollider"
      >
        SuperCollider
      </option>
      <option
        className="c5"
        value="Swift"
      >
        Swift
      </option>
      <option
        className="c5"
        value="SystemVerilog"
      >
        SystemVerilog
      </option>
      <option
        className="c5"
        value="TI Program"
      >
        TI Program
      </option>
      <option
        className="c5"
        value="Tcl"
      >
        Tcl
      </option>
      <option
        className="c5"
        value="TeX"
      >
        TeX
      </option>
      <option
        className="c5"
        value="Terra"
      >
        Terra
      </option>
      <option
        className="c5"
        value="Turing"
      >
        Turing
      </option>
      <option
        className="c5"
        value="TypeScript"
      >
        TypeScript
      </option>
      <option
        className="c5"
        value="UnrealScript"
      >
        UnrealScript
      </option>
      <option
        className="c5"
        value="VCL"
      >
        VCL
      </option>
      <option
        className="c5"
        value="VHDL"
      >
        VHDL
      </option>
      <option
        className="c5"
        value="Vala"
      >
        Vala
      </option>
      <option
        className="c5"
        value="Verilog"
      >
        Verilog
      </option>
      <option
        className="c5"
        value="Vim script"
      >
        Vim script
      </option>
      <option
        className="c5"
        value="Visual Basic"
      >
        Visual Basic
      </option>
      <option
        className="c5"
        value="Volt"
      >
        Volt
      </option>
      <option
        className="c5"
        value="Vue"
      >
        Vue
      </option>
      <option
        className="c5"
        value="WebAssembly"
      >
        WebAssembly
      </option>
      <option
        className="c5"
        value="X10"
      >
        X10
      </option>
      <option
        className="c5"
        value="XC"
      >
        XC
      </option>
      <option
        className="c5"
        value="XQuery"
      >
        XQuery
      </option>
      <option
        className="c5"
        value="XSLT"
      >
        XSLT
      </option>
      <option
        className="c5"
        value="Yacc"
      >
        Yacc
      </option>
      <option
        className="c5"
        value="Zephir"
      >
        Zephir
      </option>
      <option
        className="c5"
        value="cpp"
      >
        cpp
      </option>
      <option
        className="c5"
        value="eC"
      >
        eC
      </option>
      <option
        className="c5"
        value="nesC"
      >
        nesC
      </option>
      <option
        className="c5"
        value="ooc"
      >
        ooc
      </option>
      <option
        className="c5"
        value="wdl"
      >
        wdl
      </option>
      <option
        className="c5"
        value="wisp"
      >
        wisp
      </option>
      <option
        className="c5"
        value="xBase"
      >
        xBase
      </option>
    </select>
    <input
      aria-label="Search for repositories"
      className="c6"
      onChange={[Function]}
      onKeyPress={[Function]}
      required={true}
      type="text"
      value=""
    />
    <button
      className="c7"
      onClick={[Function]}
    >
      <img
        alt="Search"
        className="c2"
        src="test-file-stub"
      />
    </button>
  </form>,
  @media screen and (min-width:40rem) {
  .c0 {
    display: flex;
    flex-flow: row wrap;
    justify-content: space-around;
    margin: 0 1.5rem;
  }
}

<div
    className="c0"
  />,
  .c0 {
  cursor: pointer;
  background: #fff;
  border: 0;
  border-radius: 0.5rem;
  font-size: 1.2rem;
  color: #000;
  text-transform: uppercase;
}

.c1 {
  margin: auto;
  padding: 0;
  box-shadow: 0px 3px 5px 1px rgba(217,217,217,0.82);
  width: 0;
  height: 0;
  overflow: hidden;
}

@media screen and (min-width:40rem) {

}

<footer>
    <button
      className="c0 c1"
      onClick={[Function]}
    >
      Show more
    </button>
  </footer>,
]
`;
